src/engine/experimental/eip7547.md

```diff
@@ -0,0 +1,100 @@
+# Engine API -- EIP7547
+
+Engine API changes introduced in EIP7547.
+
+This specification is based on and extends [Engine API - Cancun](./cancun.md) specification.
+
+## Table of contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Structures](#structures)
+  - [PayloadAttributesV4](#payloadattributesv4)
+- [Methods](#methods)
+  - [engine_newPayloadV4](#engine_newpayloadv4)
+    - [Request](#request)
+    - [Response](#response)
+    - [Specification](#specification)
+  - [engine_forkchoiceUpdatedV4](#engine_forkchoiceupdatedv4)
+    - [Request](#request-1)
+    - [Response](#response-1)
+    - [Specification](#specification-1)
+  - [engine_getInclusionListV1](#engine_getinclusionlistv1)
+    - [Request](#request-2)
+    - [Response](#response-2)
+    - [Specification](#specification-2)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Structures
+
+### PayloadAttributesV4
+
+This structure has the syntax of [`PayloadAttributesV3`](./cancun.md#payloadattributesv3) and appends the new fields: `inclusionListTransactions`.
+
+- `timestamp`: `QUANTITY`, 64 Bits - value for the `timestamp` field of the new payload
+- `prevRandao`: `DATA`, 32 Bytes - value for the `prevRandao` field of the new payload
+- `suggestedFeeRecipient`: `DATA`, 20 Bytes - suggested value for the `feeRecipient` field of the new payload
+- `withdrawals`: `Array of WithdrawalV1` - Array of withdrawals, each object is an `OBJECT` containing the fields of a `WithdrawalV1` structure.
+- `parentBeaconBlockRoot`: `DATA`, 32 Bytes - Root of the parent beacon block.
+- `inclusionListTransations`: `Array of DATA` - Array of transaction objects, each object is a byte list (`DATA`) representing `TransactionType || TransactionPayload` or `LegacyTransaction` as defined in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718)
+
+## Methods
+
+### engine_newPayloadV4
+
+#### Request
+
+Refer to the request for [`engine_newPayloadV3`](./cancun.md#engine_newpayloadv3).
+
+#### Response
+
+Refer to the response for [`engine_newPayloadV3`](./cancun.md#engine_newpayloadv3).
+
+#### Specification
+
+This method follows the same specification as [`engine_newPayloadV3`](./cancun.md#engine_newpayloadv3) with the addition of the following:
+
+1. Client software **MUST** check that provided inclusion list summary is satisfied. Each entry in the summary must have a transaction in the parent payload or the existing block payload from that address.
```

Comment:
This references an inclusion list summary, but the `newPayloadV3` request does not contain an inclusion list summary. needs either a new parameter (like v3 did with `versioned_hashes`), or `ExecutionPayloadV4` should be defined. New parameter _seems_ more correct, since it will be checked alongside the payload?

Reply:
revising